pulser/simulation/simresults.py

```diff
@@ -204,7 +212,7 @@ def sample_final_state(self, meas_basis=None, N_samples=1000):
                     if v == '0':
                         ind.append(ex_one)
                     else:
-                        ind.append(one_state)
+                        ind.append(cast(slice, one_state))
```

Comment:
This is not a slice, it's an `int`. What's the error you're getting?

Reply:
Getting `error: Argument 1 to "append" of "list" has incompatible type "int"; expected "slice"`

Reply:
This means that the type of `ind` is being implicitly thought to be `list[slice]`, because of line 213. You should annotate it explicitly to be `list[Union[int, slice]]`

Reply:
Just a reminder to address this